packages/kbn-management/settings/utilities/mocks/settings.mock.ts

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { KnownTypeToMetadata, SettingType } from '@kbn/management-settings-types';
+
+const defaults = {
+  requiresPageReload: false,
+  readonly: false,
+};
+
+type Settings = {
+  [key in SettingType]: KnownTypeToMetadata<key>;
+};
+
+/**
+ * A utility function returning a representative set of UiSettings.
+ * @param requirePageReload The value of the `requirePageReload` param for all settings.
+ */
+export const getSettingsMock = (requirePageReload: boolean = false): Settings => {
+  if (requirePageReload) {
```

Comment:
nit: If we call `getSettingsMock` with `requirePageReload` set to true, this will permanently change the `requiresPageReload` property in the `defaults` object and we'll not be able to set `requiresPageReload` back to `false`.

```suggestion
const defaults = {
  requiresPageReload: requirePageReload,
  readonly: false,
};
```


Also, shall we import this function in `settings/components/form/form.test.tsx` and `settings/components/form/storybook/form.stories.tsx`, and remove the redundant `components/form/mocks/settings.ts`?